File: Barath/src/11_Count_Vowels/count-vowels.test.ts
// countVowels.test.ts

import { countVowels } from './count-vowels';

describe('countVowels', () => {
    
    // Test with a string containing vowels
    test('should count vowels in "Hello, World!"', () => {
        const input = "Hello, World!";
        const result = countVowels(input);
        expect(result).toBe(3);
    });

    // Test with a string containing no vowels
    test('should return 0 when there are no vowels', () => {
        const input = "bcdfg";
        const result = countVowels(input);
        expect(result).toBe(0);
    });

    // Test with an empty string
    test('should return 0 for an empty string', () => {
        const input = "";
        const result = countVowels(input);
        expect(result).toBe(0);
    });

    // Test with a string containing only vowels
    test('should count vowels when only vowels are present', () => {
        const input = "aeiouAEIOU";
        const result = countVowels(input);
        expect(result).toBe(10);
    });
});

File: Bipin/011_count_vowels/count_vowels_test.go
package main

import (
	"testing"
)

// Test function for countVowels
func TestCountVowels(t *testing.T) {
	// Test case 1: String with vowels
	result := CountVowels("Hello, World!")
	expected := 3
	if result != expected {
		t.Errorf("Test Case 1 Failed: expected %d, got %d", expected, result)
	}

	// Test case 2: String with no vowels
	result = CountVowels("xyz123")
	expected = 0
	if result != expected {
		t.Errorf("Test Case 2 Failed: expected %d, got %d", expected, result)
	}

	// Test case 3: Empty string
	result = CountVowels("")
	expected = 0
	if result != expected {
		t.Errorf("Test Case 3 Failed: expected %d, got %d", expected, result)
	}

	// Test case 4: String with only vowels
	result = CountVowels("aeiouAEIOU")
	expected = 10
	if result != expected {
		t.Errorf("Test Case 4 Failed: expected %d, got %d", expected, result)
	}
}
